inference-engine/thirdparty/clDNN/src/graph_optimizer/add_required_reorders.cpp

```diff
@@ -51,28 +51,9 @@ void add_required_reorders::run(program& p) {
             continue;  // only nodes with dependencies
         if (usr->is_type<data>())
             continue;
+        if (usr->type()->does_an_implementation_exist(*usr))
+            continue;
 
-        if (usr->type()->does_an_implementation_exist(*usr)) {
-            if (usr->get_preferred_impl_type() != impl_types::onednn) {
-                continue;
-            } else {
-                // oneDNN doesn't support padded memory, so add reorder directly if needed
-                for (size_t i = 0; i < usr->get_dependencies().size(); i++) {
-                    auto& input = usr->get_dependency(i);
-                    if (!input.is_in_data_flow() || input.is_constant())
-                        continue;
-
-                    if (static_cast<bool>(input.get_output_layout().data_padding)) {
-                        cldnn::layout layout_wo_padding = input.get_output_layout();
-                        layout_wo_padding.data_padding = cldnn::padding{};
-                        auto new_reorder = std::make_shared<reorder>(input.id() + "_padding_reorder_" + usr->id(), input.id(), layout_wo_padding);
-                        auto& new_reorder_node = p.get_or_create(new_reorder);
-                        p.add_intermediate(new_reorder_node, *usr, i);
-                    }
-                }
-                continue;
-            }
-        }
```

Comment:
This is for both f-padding and spatial padding. Restriction for spatial-padding should remain as-is.

Reply:
Okey, I see, updated it.